functions/src/alchemy_webhook.ts

```diff
@@ -42,9 +43,29 @@ type AddressNotification = {
   event: AddressEvent;
 };
 
+function isValidSignatureForStringBody(
+  body: string, // must be raw string body, not json transformed version of the body
+  signature: string, // your "X-Alchemy-Signature" from header
+  signingKey: string // taken from dashboard for specific webhook
+): boolean {
+  const hmac = crypto.createHmac("sha256", signingKey); // Create a HMAC SHA256 hash using the signing key
+  hmac.update(body, "utf8"); // Update the token hash with the request body using utf8
+  const digest = hmac.digest("hex");
+  return signature === digest;
+}
+
 export const alchemyWebhook = functions.https.onRequest(async (req, res) => {
   const body = req.body;
+  const headerKey = req.headers["x-alchemy-signature"] as string;
+  const signingKey = functions.config().alchemy.signingkey;
```

Comment:
Can you update `.github/workflows/firebase-functions.yml` to also set this secret? Otherwise it'll get overridden on deploy.